.github/workflows/docs.yml

```diff
@@ -0,0 +1,39 @@
+name: Test Documentation Build
+on:
+  pull_request:
+    branches:
+      - master
+      - v14
+      - doc-fixes
+  push:
+    branches: 
+      - master
+      - v14
+      - doc-fixes
+
+jobs:
+  test-sphinx-build:
+    name: test-sphinx-build
+    runs-on: ${{matrix.os}}
+    strategy:
+      matrix:
+        python-version: [3.7]
+        os: [ubuntu-latest]
+      fail-fast: False
+    steps:
+      - uses: actions/checkout@v2
+      - name: Initialize vendored libs
+        run:
+          git submodule update --init --recursive
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -W ignore -m pip install --upgrade pip
+          python -W ignore -m pip install -r requirements.txt
+          python -W ignore -m pip install -r requirements-dev.txt
+          python -W ignore -m pip install -r docs/requirements-docs.txt
+      - name: Build docs
+        run: sphinx-build docs/source docs/build/html -W --keep-going
```

Comment:
We can add the argument `j auto` here. https://www.sphinx-doc.org/en/master/man/sphinx-build.html#cmdoption-sphinx-build-j.

I'm adding it to the makefile in #2841

Reply:
"this option should be considered _experimental_" ... Does it speed up much for you?

Reply:
by a factor of 10x or more

Reply:
o.0 for me, it didn't speed up at all 😄 but if it helps for you, let's add it - we can revert if there are problems and the feature has been around for 3 major versions now so probably not _that_ experimental anymore 🤔

Reply:
it really speeds it up when you run `make -C docs clean` and then `make -C docs html`

Reply:
jup, tried that. but it's probably also dependent on how many other processes are running on the machine etc …